tests/postgres_tests/test_constraints.py

```diff
@@ -0,0 +1,181 @@
+                timezone.datetime(year=2018, month=6, day=30).date()
+            ),
+            room=self.room102,
+        )
+        self.assertIsNotNone(reservation.pk)
+
+    def test_where_constraint(self):
+        reservation = Reservation.objects.create(
+            timespan=DateRange(
+                timezone.datetime(year=2018, month=6, day=25).date(),
+                timezone.datetime(year=2018, month=6, day=27).date()
+            ),
+            room=self.room102,
+            cancelled=True,
+        )
+        self.assertIsNotNone(reservation.pk)
+
+    def test_overlap_banned(self):
+        with self.assertRaises(IntegrityError):
+            Reservation.objects.create(
+                timespan=DateRange(
+                    timezone.datetime(year=2018, month=6, day=25).date(),
+                    timezone.datetime(year=2018, month=6, day=27).date()
+                ),
+                room=self.room102,
+                cancelled=False,
+            )
+
+    def test_update_(self):
+        reservation = Reservation.objects.create(
+            timespan=DateRange(
+                timezone.datetime(year=2018, month=6, day=25).date(),
+                timezone.datetime(year=2018, month=6, day=27).date()
+            ),
+            room=self.room102,
+            cancelled=True,
+        )
+        with self.assertRaises(IntegrityError):
+            reservation.cancelled = False
+            reservation.save()
+
+    def test_repr(self):
+        constraint = ExclusionConstraint(
+            name='exclude_overlapping',
+            expressions=[(F('timespan'), '&&'), (F('room'), '=')],
+        )
+        self.assertEqual(
+            repr(constraint),
+            "<ExclusionConstraint: Using GIST, Expressions=[(F(timespan), '&&'), (F(room), '=')]>"
+        )
+        # without the additional condition
+        constraint = ExclusionConstraint(
+            name='exclude_overlapping',
+            expressions=[(F('timespan'), '&&'), (F('room'), '=')],
+            condition=Q(cancelled=False),
+        )
+        self.assertEqual(
+            repr(constraint),
+            "<ExclusionConstraint: Using GIST, "
+            "Expressions=[(F(timespan), '&&'), (F(room), '=')], "
+            "Where=(AND: ('cancelled', False))>"
+        )
+
+    def test_eq(self):
+        constraint = ExclusionConstraint(
+            name='exclude_overlapping',
+            expressions=[(F('timespan'), '&&'), (F('room'), '=')],
+            condition=Q(cancelled=False),
+        )
+        self.assertEqual(constraint, constraint)
+
+    def test_not_eq(self):
+        first = ExclusionConstraint(
+            name='exclude_overlapping',
+            expressions=[(F('timespan'), '&&'), (F('room'), '=')],
+            condition=Q(cancelled=False),
+        )
+        second = ExclusionConstraint(
+            name='exclude_overlapping',
+            expressions=[(F('timespan')), (F('room'), '=')],
```

Comment:
Is the first element here meant to be a 1-tuple? You're missing the required comma.

Reply:
No, it's a mistake. Thanks for noticing.